src/Mono.Android/Android.Security/KeyChain.cs

```diff
@@ -0,0 +1,94 @@
+#nullable enable
+using System;
+using System.Runtime.Versioning;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+
+namespace Android.Security
+{
+	public partial class KeyChain
+	{
+		[SupportedOSPlatform("android14.0")]
+		public static X509Certificate2? GetCertificateWithPrivateKey (Android.Content.Context context, string alias)
+		{
+			var privateKey = KeyChain.GetPrivateKey (context, alias);
+			if (privateKey is null) {
+				return null;
+			}
+
+			var chain = KeyChain.GetCertificateChain (context, alias);
+			if (chain is null) {
+				return null;
+			}
+
+			var privateKeyEntry = new Java.Security.KeyStore.PrivateKeyEntry (privateKey, chain);
+			return new X509Certificate2 (privateKeyEntry.Handle);
+		}
+
+		[SupportedOSPlatform("android23.0")]
+		public static async Task<string?> ChoosePrivateKeyAlias (
```

Comment:
`async` methods should have an `Async` suffix, [by convention](https://learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/task-asynchronous-programming-model):

> * The name of an async method, by convention, ends with an "Async" suffix.